### PR TITLE
refactoring of `azurerm_network_interface` and `azurerm_availability_set`

### PR DESCRIPTION
**Pull request by tombuildsstuff**
- Exporting the list of Private IP Addresses assigned to the VM
- Refactoring the `azurerm_network_interface` and `azurerm_availability_set` resources

### CODE REVIEW

File: azurerm/resource_arm_network_interface.go [hunk 1]
```diff
@@ -151,6 +151,13 @@ func resourceArmNetworkInterface() *schema.Resource {
 				Computed: true,
 			},
 
+			"private_ip_addresses": {
+				Type:     schema.TypeSet,
```
Comment (radeksimko) [inline azurerm/resource_arm_network_interface.go]: I'm thinking this _may_ theoretically cause some troubles when referencing the field as set has computed indexes? 🤔

**Review by radeksimko — APPROVED** [review]
Just one minor question, otherwise LGTM. I'm assuming all related acceptance tests are passing.

File: azurerm/resource_arm_network_interface.go (hunk 1 above)
Comment (tombuildsstuff) [reply azurerm/resource_arm_network_interface.go]: Good point.. I'll look into switching this over to a list

Comment (tombuildsstuff) [reply azurerm/resource_arm_network_interface.go]: Updated

**Comment by tombuildsstuff** [comment]
Tests pass:

screen shot 2017-10-11 at 15 12 07

**Pull request merged**